mulog-prometheus/src/com/brunobonacci/mulog/publishers/prometheus/collector.clj

```diff
@@ -0,0 +1,148 @@
+(ns com.brunobonacci.mulog.publishers.prometheus.collector
+  (:import [io.prometheus.client
+            SimpleCollector
+            SimpleCollector$Builder
+            Counter
+            Counter$Child
+            Gauge
+            Gauge$Child
+            Histogram
+            Histogram$Child
+            Summary
+            Summary$Builder
+            Summary$Child]))
+
+(def ^:private summary-quantiles-default [[0.5   0.001]
+                                          [0.9   0.001]
+                                          [0.95  0.001]
+                                          [0.99  0.001]
+                                          [0.999 0.001]])
+
+(def ^:private metric-suffix
+  {:counter    "cntr"
+   :gauge      "gauge"
+   :histogram  "hstgm"
+   :summary    "smry"})
+
+
+
+(defprotocol ChildWithLabels
+  (child-with-labels [t label-values]))
+
+(defprotocol Increment
+  (increment [t]))
+
+(defprotocol SetValue
+  (set-value [t value]))
+
+(defprotocol ObserveValue
+  (observe-value [t value]))
+
+(extend-type SimpleCollector
+  ChildWithLabels
+  (child-with-labels [t label-values] (.labels t label-values)))
+
+(extend-type Counter$Child
+  Increment
+  (increment [t] (.inc t)))
+
+(extend-type Gauge$Child
+  SetValue
+  (set-value [t value] (.set t value)))
+
+(extend-type Histogram$Child
+  ObserveValue
+  (observe-value [t value] (.observe t value)))
+
+(extend-type Summary$Child
+  ObserveValue
+  (observe-value [t value] (.observe t value)))
+
+
+
+
+(defn- simple-collector-builder
+  [builder-constructor
+   {:keys [name namespace description label-keys]}]
+  (-> ^SimpleCollector$Builder
+   (builder-constructor)
```

Comment:
It is preferable that the tag `^SimpleCollector$Builder` is on the same line as the entity being tagged.

Reply:
Noted, thank you